src/github.com/stellar/horizon/actions_trade.go

```diff
@@ -63,14 +71,50 @@ func (action *TradeIndexAction) loadRecords() {
 	action.Err = trades.Page(action.PagingParams).Select(&action.Records)
 }
 
+// loadLedgers collects the unique ledgers referenced in the loaded trades and loads the details for each.
+func (action *TradeIndexAction) loadLedgers() {
+	if len(action.Records) == 0 {
+		return
+	}
+
+	ledgerSequences := make([]interface{}, len(action.Records))
```

Comment:
Because the underlying where call accepts `[]interface{}` and `[]int32` cannot be casted to `[]interface{}`.  You'd have to manually create a populate a new slice within `LedgersBySequence`.  Since `LedgersBySequence` is effectively an internal only method, I'm fine with having the leniency... you'll still get a good error message if you pass an invalid value, but any numbers and strings work as expected.